Week3/homework/step2-7.js

```diff
@@ -20,4 +20,15 @@ f2(y);
 
 console.log(y);
 
-// Add your explanation as a comment here
+// Explanation:
+
+/** Pass by value is whereby a function is called by directly passing the value of
+ * the variable as the argument. The variable passed from outside the function is not
+ * affected as a result of changing the argument inside the function. In this case,
+ * the value of 'x' is not changed in the f1 function.
+ *
+ * While pass by reference is whereby a function is called directly by passing the
+ * reference or address of the variable as the argument. Changing the argument inside
+ * the function affects the variable passed in from outside the function as in the case
+ * of f2 function above.
+ */
```

Comment:
Okay, the key point is passing by value as opposed to passing by reference.

Reply:
Hi Jim, 
All corrections have been made. It should be fine now